quickwit/quickwit-aws/src/lib.rs

```diff
@@ -98,11 +99,15 @@ impl ExtendedChainProvider {
 #[async_trait]
 impl ProvideAwsCredentials for ExtendedChainProvider {
     async fn credentials(&self) -> Result<AwsCredentials, CredentialsError> {
-        if let Ok(credentials) = self.web_identity_provider.credentials().await {
-            return Ok(credentials);
+        match self.web_identity_provider.credentials().await {
+            Ok(credentials) => return Ok(credentials),
+            Err(error) => {
+                warn!(err=?error, "Failed to find AWS credentials from `WebIdentityProvider`.")
```

Comment:
We will always display this warning for users who rely on the "regular" `ChainProvider`. Let's store the error from the first provider and display it only if both authentication methods fail. We can also add debug statements to display which method was elected.